pyenttec/__init__.py

```diff
@@ -209,10 +213,25 @@ def set_channel(self, chan, val):
             raise DMXAddressError("Channel index {} out of range. "
                                   "Universe size is {}."
                                   .format(chan, len(self.dmx_frame)))
+        except OverflowError:
+            raise DMXOverflowError("Channel value {} out of range. "
+                                  "DMX uses 8bit unsigned values (0-255)."
+                                  .format(chan, len(self.dmx_frame)))
+
+    def __setitem__(self, chan, val):
+        self.set_channel(chan, val)
```

Comment:
Since in a sense `__setitem__` is the basic implementation here, it probably makes more sense to move the guts of `set_channel` in here and reduce the impl of `set_channel` to
```python
def set_channel(self, chan, val):
    self[chan] = val
```

That avoids the method call overhead for each channel set operation in what is probably your most common case.

Reply:
In-fact we could actually do `__setitem__ = set_channel` or the inverse to avoid the overhead in either case. Didn't do that originally cause I think it's less clear, but if we're gonna worry about reducing method calls that's probably a better way to do it.

Reply:
I'm of the belief that the more descriptive method name should contain the logic, and that `__setitem__` is more of a shortcut. So will push with the `__setitem__ = set_channel` optimisation for now. Can overrule of course if you disagree

Reply:
`set_channel` is a remnant of the API that I forked originally; I don't think it's necessary at all but I don't want to remove it for backwards-compatibility.  Assigning to the attribute looks good to me.